internal/gitlab/gitlab.go

```diff
@@ -874,6 +874,49 @@ func LabelDelete(project, name string) error {
 	return err
 }
 
+// BranchGet gets a specific named branch object from the project
+func BranchGet(project string, name string) (*gitlab.Branch, error) {
+	opts := &gitlab.ListBranchesOptions{
+		Search: &name,
+	}
+
+	branches, _ := BranchList(project, opts)
+	switch len(branches) {
+	case 1:
+		return branches[0], nil
+	case 0:
+		return nil, errors.Errorf("Branch '%s' not found", name)
+	default:
+		return nil, errors.Errorf("Branch '%s' is ambiguous", name)
+	}
+}
+
+// BranchList get all branches from the project that somehow matches the
+// name requested
```

Comment:
```suggestion
// requested options
```

maybe? It's used for matching on a name in BranchGet, but the function itself is more generic.

Reply:
Yep, well pointed.